### DISCUSSION TITLE
Share your [pid-vid] for your controller

### DISCUSSION THREAD

**Discussion by o0Zz** [o0Zz/sys-con#12] [General]
Hello there,

You have added a new controller to sys-con and want to share it with others, this is the place to be ;)
(These controllers will be merged regularly - so they will be part of the main project and you will not have to maintain them)

Have fun !

**Comment by onLook3rEk0r** [comment]
can i connect my Dualsens(PS5)&dualsense Edge controller？

**Reply by o0Zz** [reply]
You could try to see if it works. I don't have ps5 controller to tests and I don't know if the ps5 controller natively works on a PC (without driver). If they do then it should works with sys-con. Move your question to Q&A please

**Comment by texita** [comment]
![image](https://github.com/o0Zz/sys-con/assets/1828679/0c4c7c04-4750-4eba-bcf4-18df00005cfc)
![image](https://github.com/o0Zz/sys-con/assets/1828679/aa8820b0-1d48-4491-8ccc-19627227d484)

https://www.powera.com/p/nintendo/nintendo-switch/controllers/wired/wired-controller-for-nintendo-switch-white-1517033-01/

[20d6-a711] ;PowerA Core (Plus) Wired Controller SKU: 1517033-01
b=2
a=3
x=4
y=1
l=5
r=6
zl=7
zr=8
minus=9
plus=10
capture=14
home=13
lstick_click=11
rstick_click=12

**Comment by ness151** [comment]
https://www.amazon.com/HORI-Real-Arcade-Pro-PlayStation-4/dp/B00SULMRI4
[0f0d-008b] ; Hori Real Arcade Pro 4
b=2
a=3
x=4
y=1
l=5
r=6
zl=7
zr=8
minus=12
plus=10
capture=9
home=13
lstick_click=
rstick_click=
![20240727_082823](https://github.com/user-attachments/assets/42a1785f-8b31-4990-9939-d62d67d7fdab)

https://www.amazon.com/Qanba-Drone-PlayStation-4/dp/B01I0GEDEY
[2c22-2003] ;Qanba Drone 1
b=1
a=2
x=4
y=3
l=5
r=6
zl=7
zr=8
minus=9
plus=10
capture=11
home=13
lstick_click=
rstick_click=
![20240727_082843](https://github.com/user-attachments/assets/75aec5a5-6766-4dbf-b850-51b0b9004d67)

I use these exclusively for fighting games on the switch, hence why the sticks aren't mapped to anything

**Reply by o0Zz** [reply]
Thank you for this share, it will be added to next version. What you means by "Why the sticks aren't mapped" ? I assume, your sticks are properly working, correct ?

**Reply by ness151** [reply]
No, as you will see in the button input I have lstick_click=
rstick_click=

Empty. My inputs are all working correctly.

**Reply by o0Zz** [reply]
Yes, that's expected, in fact you can remove those 2 lines, it won't change anything. If you look in the [default] section you will find all the possible inputs you can use, there are about 30 and most of the time you will not need to bind all of them. Regarding rstick_click/lstick_click, these are the buttons under the sticks (when you press a stick), but as you mentioned, there is no similarity on arcade controllers, most of the time you can't click on the stick, so i'm not surprise these binding are not used.

**Comment by tirom97121** [comment]
![20240731_190236](https://github.com/user-attachments/assets/dfd4ca58-1f5e-43bd-846e-88cf6e7e8ecc)

Hello everybody, i want to set up this Nintendo 64 adapter on the Switch. Can someone help me?"

**Reply by o0Zz** [reply]
Hi there is a complete guide here: https://github.com/o0Zz/sys-con?tab=readme-ov-file#configure-a-controller
and if you meet any issue, there is also a troubleshooting with most of issue you might meet https://github.com/o0Zz/sys-con?tab=readme-ov-file#configure-a-controller

**Reply by tirom97121** [reply]
OK, thanks a lot, i made this settings:

[0E8F-3013] ;N64 PC adapter
controller_type=tarragon
B=3
A=2
L=7
R=8
ZL=9
plus=10
dpad_up=13
dpad_right=14
dpad_down=15
dpad_left=16
simulate_home=ZL+plus
right_stick_x=-Rz
right_stick_y=Z

could you help me to calibrate left stick?
In sys-con.nro on my switch, all controllers up to 32000 -32000, my n64 controler 20000 -20000 max when i push up down left right.

Thanks a lot.

**Reply by o0Zz** [reply]
Thank you for this share.
Could you try to "Calibrate" your stick using the calibrate feature included in the switch and see if it solve your issue. (Settings -> Controller & sensors -> Calibrate sticks). Let me know if it help.

**Reply by o0Zz** [reply]
I think it will change nothing to your problem, but probably `controller_type=n64` would better match your use case. Isn't it ?

**Reply by tirom97121** [reply]
Same problem with n64 type, in this mode, C-Pad doesn't works too .

I tried to calibrate in switch settings, left stick make a Cube, a square with others controllers
Sorry for my bad english, i put a GIF to explain.
![20240801_233624](https://github.com/user-attachments/assets/098f414b-1678-4f9a-986e-460e16387d4d)

**Reply by o0Zz** [reply]
Please open a ticket that describe this problem and attach sys-con logs we will discuss on it in a dedicated ticket.

**Comment by oh-habeeb** [comment]
[https://www.amazon.com.au/Hyperkin-Controller-Adapter-Compatible-Nintendo/dp/B082N7K8QS](https://www.amazon.com.au/Hyperkin-Controller-Adapter-Compatible-Nintendo/dp/B082N7K8QS)
```
[20d6-a710] ;Hyperkin N64 Adapter (Switch mode)
b=2
a=3
l=5
r=6
zl=7
plus=10
rstick_left=-Z
rstick_right=Z
rstick_up=Rz
rstick_down=-Rz
```

This adapter works out of the box on a normal switch without sys-con, however if you do have sys-con, the configuration is changed so you need the above. My use case for sys-con is to change buttons so I can use minus for NSO, otherwise it has no minus or home and is just a normal N64 controller. I don't know if `controller_type=n64` would be beneficial, as I couldn't find any documentation telling me what that actually does.
My personal config has 
```
minus=32
simulate_home=minus+plus
```
which changes Dpad up to minus and gives ability for home button.
Works like a charm so far.

**Reply by o0Zz** [reply]
Thank you for this share, I included it in sys-con default config - I just used this "simulate_home=zl+plus" to match other n64 adapters ;)

**Comment by Craftyawesome** [comment]
https://www.amazon.com/Buffalo-iBuffalo-Classic-USB-Gamepad/dp/B002B9XB0E
```
[0583-2060] ;Buffalo snes usb
b=2
a=1
x=3
y=4
l=5
r=6
minus=7
plus=8
```

**Reply by o0Zz** [reply]
Thanks for this share

**Comment by l378500947** [comment]
https://www.amazon.com/-/zh/dp/B09FDCBW6W
[0d22-0c31] ; MSI GC30 V2 USB cable
b=2
a=3
x=4
y=1
l=5
r=6
zl=7
zr=8
minus=9
plus=10
lstick_click=11
rstick_click=12

**Reply by l378500947** [reply]
Works fine with USB cable, not with a wireless adapter:
[0d22-0c33] ; Automatically added MSI
profile=xboxone

I don't know how to get the wireless adapter to work。

**Comment by a0193143** [comment]
[0d22-0c20] ; MSI GC20
b=2
a=3
x=4
y=1
l=5
r=6
zl=7
zr=8
minus=9
plus=10
lstick_click=11
rstick_click=12

**Reply by o0Zz** [reply]
Thank you for this share I will add it to the main

**Comment by Canpake59** [comment]
Anyone here manage to get the steam controller 2026 to work? Is there extra setup needed to get it to work?
